### PR TITLE
RocksdbMavenPomCache - Delete non-deserializable cache values

### PR DESCRIPTION
**Pull request by pway99**
RocksdbMavenPomCache - Delete cached values when exceptions are thrown during deserialization

### CODE REVIEW

File: rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java [hunk 1]
```diff
@@ -164,7 +158,13 @@ public RocksdbMavenPomCache(@Nullable File workspace) {
...
 
         byte[] key = serialize(repo.toString() + ":" + artifactCoordinates);
-        Optional<RawMaven> rawMavenEntry = deserializeRawMaven(cache.get(key));
+        Optional<RawMaven> rawMavenEntry = null;
+        try {
+            rawMavenEntry = deserializeRawMaven(cache.get(key));
+        } catch (Exception ex) {
+            // upon serialization exceptions delete the cache entry and force it to be re-downloaded
+            cache.database.delete(key);
+        }
```
Comment (tkvangorder) [inline rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java:167]: I think we can just change deserializeRawMan() such that it catches the serialization errors and just returns null (which is a cache miss) 

Then the logic will fall through, download the pom file, serialize it (in the new format), and then replace the one that is there.

Comment (tkvangorder) [reply rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java:167]: We can do something similar for the other deserialize methods.

**Review by tkvangorder — CHANGES_REQUESTED** [review]

**Review by tkvangorder — APPROVED** [review]
Fixed to just treat errors as a cache miss.

**Comment by tkvangorder** [comment]
Good catch Pat, thanks!

**Comment by pway99** [comment]
Thanks @tkvangorder

**Pull request merged**